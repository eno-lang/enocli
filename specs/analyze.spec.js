const analyze = require('../lib/analyze');
const path = require('path');

const DOCUMENT_PATH = path.join(__dirname, 'fixtures/document.eno');

describe('analyze', () => {
  it('builds an abstract syntax tree from a document', () => {
    const astDocument = analyze(DOCUMENT_PATH);

    expect(astDocument).toMatchSnapshot();
  });
});
