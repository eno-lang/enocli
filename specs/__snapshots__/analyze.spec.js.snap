// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`analyze builds an abstract syntax tree from a document 1`] = `
Object {
  "elementRequired": true,
  "elementType": "section",
  "elements": Array [
    Object {
      "elementRequired": true,
      "elementType": "empty",
      "internalKey": "empty",
      "key": "empty",
      "parent": [Circular],
    },
    Object {
      "elementRequired": true,
      "elementType": "field",
      "internalKey": "field",
      "key": "field",
      "parent": [Circular],
      "valueRequired": true,
      "valueTypes": Array [
        Object {
          "loader": [Function],
          "name": "string",
          "placeholder": "Lorem Ipsum (e.g.)",
        },
      ],
    },
    Object {
      "elementRequired": true,
      "elementType": "fieldset",
      "entries": Array [
        Object {
          "elementRequired": true,
          "elementType": "entry",
          "internalKey": "entry",
          "key": "entry",
          "parent": [Circular],
          "valueRequired": true,
          "valueTypes": Array [
            Object {
              "loader": [Function],
              "name": "string",
              "placeholder": "Lorem Ipsum (e.g.)",
            },
          ],
        },
      ],
      "internalKey": "fieldset",
      "key": "fieldset",
      "parent": [Circular],
    },
    Object {
      "elementRequired": true,
      "elementType": "list",
      "internalKey": "list",
      "key": "list",
      "parent": [Circular],
      "valueTypes": Array [
        Object {
          "loader": [Function],
          "name": "string",
          "placeholder": "Lorem Ipsum (e.g.)",
        },
      ],
      "valuesRequired": true,
    },
    Object {
      "elementRequired": true,
      "elementType": "section",
      "elements": Array [
        Object {
          "elementRequired": true,
          "elementType": "field",
          "internalKey": "field",
          "key": "field",
          "parent": [Circular],
          "valueRequired": true,
          "valueTypes": Array [
            Object {
              "loader": [Function],
              "name": "string",
              "placeholder": "Lorem Ipsum (e.g.)",
            },
          ],
        },
      ],
      "internalKey": "section",
      "key": "section",
      "parent": [Circular],
    },
  ],
  "internalKey": "document",
}
`;

exports[`analyze builds an abstract syntax tree from a section-heavy document 1`] = `
Object {
  "elementRequired": true,
  "elementType": "section",
  "elements": Array [
    Object {
      "elementRequired": true,
      "elementType": "section",
      "elements": Array [
        Object {
          "elementRequired": true,
          "elementType": "section",
          "elements": Array [
            Object {
              "elementRequired": true,
              "elementType": "section",
              "elements": Array [],
              "internalKey": "section_depth_3",
              "key": "section (depth: 3)",
              "parent": [Circular],
            },
          ],
          "internalKey": "section_depth_2",
          "key": "section (depth: 2)",
          "parent": [Circular],
        },
      ],
      "internalKey": "section_depth_1",
      "key": "section (depth: 1)",
      "parent": [Circular],
    },
    Object {
      "elementRequired": true,
      "elementType": "section",
      "elements": Array [
        Object {
          "elementRequired": true,
          "elementType": "section",
          "elements": Array [
            Object {
              "elementRequired": true,
              "elementType": "section",
              "elements": Array [],
              "internalKey": "section_depth_3",
              "key": "section (depth: 3)",
              "parent": [Circular],
            },
          ],
          "internalKey": "section_depth_2",
          "key": "section (depth: 2)",
          "parent": [Circular],
        },
        Object {
          "elementRequired": true,
          "elementType": "section",
          "elements": Array [
            Object {
              "elementRequired": true,
              "elementType": "section",
              "elements": Array [],
              "internalKey": "section_depth_3",
              "key": "section (depth: 3)",
              "parent": [Circular],
            },
          ],
          "internalKey": "section_depth_3",
          "key": "section (depth: 2)",
          "parent": [Circular],
        },
      ],
      "internalKey": "section_depth_2",
      "key": "section (depth: 1)",
      "parent": [Circular],
    },
  ],
  "internalKey": "document",
}
`;
